src/Symfony/Component/Config/Definition/Builder/NodeBuilder.php

```diff
@@ -143,8 +143,8 @@ public function end()
     /**
      * Creates a child node.
      *
-     * @param string $name The name of the node
-     * @param string $type The type of the node
+     * @param string|null $name The name of the node
```

Comment:
It does not make sense to change this when you don't change all the other places as well like https://github.com/ro0NL/symfony/blob/2d72b61613131160d2f4716f71bae7a55a9f4153/src/Symfony/Component/Config/Definition/Builder/NodeDefinition.php#L39 and https://github.com/ro0NL/symfony/blob/2d72b61613131160d2f4716f71bae7a55a9f4153/src/Symfony/Component/Config/Definition/Builder/NodeBuilder.php#L52

Reply:
Also what does a node without a name do?

Reply:
Ok my bad.. it's clarified only on 4.0: https://github.com/symfony/symfony/blob/4.0/src/Symfony/Component/Config/Definition/BaseNode.php#L39

Should be backported to 2.7 as `string|null` AFAIK, this being the first instance :)

In general it's used to create a prototyped node: https://github.com/symfony/symfony/blob/4.0/src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php#L77

I think it makes sense to keep string API when using a named factory: `scalarNode('string')` vs `node(null, 'scalar')`. Which seems to be the case already; https://github.com/symfony/symfony/blob/4.0/src/Symfony/Component/Config/Definition/Builder/NodeBuilder.php#L52